src/include/daos/mem.h

```diff
@@ -45,6 +45,44 @@
 
 #include <libpmemobj.h>
 
+/** The offset of an object from the base address of the pool */
+typedef uint64_t		umem_off_t;
+#define UMOFF_NUM_FLAGS		(8)
+#define UMOFF_MAX_FLAG		(1ULL << UMOFF_NUM_FLAGS)
+#define UMOFF_FLAG_MASK		(UMOFF_MAX_FLAG - 1)
+#define UMOFF_NULL		(~0ULL & ~UMOFF_FLAG_MASK)
+#define UMOFF_IS_NULL(umoff)	(umoff >= UMOFF_NULL)
+
+/** Up to UMOFF_NUM_FLAGS bits are available to the user to mark an
+ *  invalid offset with extra information.  If these fields are
+ *  set, the pointer resulting from such an offset will be
+ *  considered NULL.
+ *
+ *  \param	offset[IN,OUT]	The value is marked invalid with the flags
+ *  \param	flags[IN]	Auxilliarly information about the invalid entry
+ */
+static inline void
+umem_off_set_invalid(umem_off_t *offset, uint64_t flags)
```

Comment:
What's the intention of the invalid flags? Seems to only be used in test functions

Reply:
See explanation in response to Liang above.   They are to support using umem_off_t for DTX where we have multiple values such as NULL (which means committed), aborted, etc.  It's not used in this patch but intended to enable future changes.